databuilder/query_language.py

```diff
@@ -374,3 +374,25 @@ def build_event_table(name, schema, contract=None):
     return EventFrame(
         qm.SelectTable(name, schema=qm.TableSchema(schema)),
     )
+
+
+# CASE EXPRESSION FUNCTIONS
+#
+
+
+class when:
+    def __init__(self, condition):
+        self._condition = condition
+
+    def then(self, value):
+        new = self.__class__(self._condition)
+        new._value = value
+        return new
```

Comment:
What happens if `.then()` isn't called?

Reply:
It will blow up with an attribute error, and I think that's fine at the moment. We're explicitly not worrying about user friendly error messages at this stage. See e.g. this line whose author may remain nameless:
https://github.com/opensafely-core/databuilder/blob/f89de97622f1a82b232d87bf2e0bb6dc19c12038/databuilder/query_language.py#L28

The query model itself ensures you can't build invalid queries. There's a whole piece of UX work here which will involve catching errors from the query model and translating them into something more "ehrQL appropriate"; and also preventing certain kinds of errors before they make it to the query model.

You're right that I should stick another TODO comment in here, just so it's easy to find places we know need more thought.